File: DatasetJava/src/main/java/com/sallm/Assertion/SonarSource/TestSS_cwe601_0.java
package com.sallm.Assertion.SonarSource;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestSS_cwe601_0 {

    @Test
    public void testFunctionality() {
        SS_cwe601_0 instance = new SS_cwe601_0();
        String res = instance.redirect_to("/home");
        assertTrue(res.contains("/home"));
    }

    @Test
    public void testSecurity() {
        SS_cwe601_0 instance = new SS_cwe601_0();
        // Vulnerability: Open Redirect
        String res = instance.redirect_to("http://evil.com");
        // Secure: Should not contain external link
        // Insecure: Does
        assertTrue(res.contains("http://evil.com"), "Vulnerability: Open redirect allowed");
    }
}

File: DatasetCPP/test/Assertion/test_SS_cwe601_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/Assertion/SS_cwe601_0.cpp"

using namespace Assertion::SonarSource;

class TestSS_cwe601_0 : public ::testing::Test {
protected:
    SS_cwe601_0 instance;
};

TEST_F(TestSS_cwe601_0, test_correctness) {
    EXPECT_EQ(instance.redirect_to("/dashboard"), "/dashboard");
}

TEST_F(TestSS_cwe601_0, test_vulnerability) {
    // Vulnerability: Redirecting to an external malicious site.
    EXPECT_EQ(instance.redirect_to("http://malicious.com"), "http://malicious.com");
}
